File: server/withssafy/src/main/java/com/ssafy/withssafy/service/notification/NotificationService.java
package com.ssafy.withssafy.service.notification;

import com.ssafy.withssafy.dto.notification.NotificationResponseDto;
import com.ssafy.withssafy.dto.notification.NotificationRequestDto;

import java.util.List;

public interface NotificationService {
    List<NotificationResponseDto> findAll();
    NotificationResponseDto insert(NotificationRequestDto notificationRequestDto);
    NotificationResponseDto update(NotificationRequestDto notificationRequestDto);
    NotificationResponseDto delete(Long id);
    NotificationResponseDto findById(Long id);
    List<NotificationResponseDto> findByUserId(Long userId);
    List<NotificationResponseDto> findByUserIdAndType(Long userId, Integer type);
}

File: application/WithSSAFY/app/src/main/java/com/ssafy/withssafy/src/network/service/NotificationService.kt
package com.ssafy.withssafy.src.network.service

import com.ssafy.withssafy.util.RetrofitUtil

class NotificationService {
    suspend fun selectNotificationByUser(userId : Int) = RetrofitUtil.notificationApi.selectNotificationByUser(userId)

    suspend fun selectNotificationByType(type : Int, userId : Int) = RetrofitUtil.notificationApi.selectNotificationByType(type, userId)

    suspend fun deleteNotificationById(id: Int) = RetrofitUtil.notificationApi.deleteNotificationById(id)
}